streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1081,8 +1081,9 @@ private void shutdown(final boolean cleanRun) {
         streamsMetrics.removeAllSensors();
     }
 
+    // visible for testing
     @SuppressWarnings("ThrowableNotThrown")
-    private void shutdownTasksAndState(final boolean cleanRun) {
+    void shutdownTasksAndState(final boolean cleanRun) {
```

Comment:
We do have places already where we use reflections; I agree there is no universal rules yet, though I think for "test-only" functions we can just make it non-private with a comment, while for "production" functions which is only accessed for testing it's better to use reflections.